tests/ghost_test.go

```diff
@@ -0,0 +1,196 @@
+	cutoverLockTimoutSeconds            = 3
+	exponentialBackoffMaxInterval       = 64
+)
+
+func newMigrationContext(config config) (*base.MigrationContext, error) {
+	migrationContext := base.NewMigrationContext()
+	migrationContext.InspectorConnectionConfig.Key.Hostname = config.host
+	migrationContext.InspectorConnectionConfig.Key.Port = config.port
+	migrationContext.CliUser = config.user
+	migrationContext.CliPassword = config.password
+	migrationContext.DatabaseName = config.database
+	migrationContext.OriginalTableName = config.table
+	migrationContext.AlterStatement = config.alterStatement
+	migrationContext.Noop = !config.execute
+	// set defaults
+	migrationContext.AllowedRunningOnMaster = allowedRunningOnMaster
+	migrationContext.ConcurrentCountTableRows = concurrentCountTableRows
+	migrationContext.HooksStatusIntervalSec = hooksStatusIntervalSec
+	migrationContext.ReplicaServerId = replicaServerID
+	migrationContext.CutOverType = base.CutOverAtomic
+
+	if migrationContext.AlterStatement == "" {
+		return nil, fmt.Errorf("alterStatement must be provided and must not be empty")
+	}
+	parser := ghostsql.NewParserFromAlterStatement(migrationContext.AlterStatement)
+	migrationContext.AlterStatementOptions = parser.GetAlterStatementOptions()
+
+	if migrationContext.DatabaseName == "" {
+		if parser.HasExplicitSchema() {
+			migrationContext.DatabaseName = parser.GetExplicitSchema()
+		} else {
+			return nil, fmt.Errorf("database must be provided and database name must not be empty, or alterStatement must specify database name")
+		}
+	}
+	if migrationContext.OriginalTableName == "" {
+		if parser.HasExplicitTable() {
+			migrationContext.OriginalTableName = parser.GetExplicitTable()
+		} else {
+			return nil, fmt.Errorf("table must be provided and table name must not be empty, or alterStatement must specify table name")
+		}
+	}
+	migrationContext.ServeSocketFile = fmt.Sprintf("/tmp/gh-ost.%s.%s.sock", migrationContext.DatabaseName, migrationContext.OriginalTableName)
+	migrationContext.SetHeartbeatIntervalMilliseconds(heartbeatIntervalMilliseconds)
+	migrationContext.SetNiceRatio(niceRatio)
+	migrationContext.SetChunkSize(chunkSize)
+	migrationContext.SetDMLBatchSize(dmlBatchSize)
+	migrationContext.SetMaxLagMillisecondsThrottleThreshold(maxLagMillisecondsThrottleThreshold)
+	migrationContext.SetDefaultNumRetries(defaultNumRetries)
+	migrationContext.ApplyCredentials()
+	if err := migrationContext.SetCutOverLockTimeoutSeconds(cutoverLockTimoutSeconds); err != nil {
+		return nil, err
+	}
+	if err := migrationContext.SetExponentialBackoffMaxInterval(exponentialBackoffMaxInterval); err != nil {
+		return nil, err
+	}
+	return migrationContext, nil
+}
+
+func TestGhostSimpleNoop(t *testing.T) {
+	const port = 13306
+	basedir := t.TempDir()
+	datadir := filepath.Join(basedir, "data")
+	if err := os.Mkdir(datadir, 0755); err != nil {
+		t.Fatal(err)
+	}
+	mysql, err := mysql.Install(basedir, datadir, "root")
```

Comment:
We're a bit overusing "mysql", some referring to package name but some referring to variable names.